src/errno.rs

```diff
@@ -1,24 +1,31 @@
+#![macro_use]
+
 extern crate libc;
+
 use std::ffi::CString;
+use std::ffi::CStr;
 use std::os::raw::c_char;
 use libc::c_int;
 use fileops::open;
 
-pub fn strerrer(errno: c_int) -> String {
+pub fn strerror(errno: c_int) -> String {
   unsafe {
-    let c_err_str = libc::strerror(errno);
-    CString::from_raw(c_err_str).into_string()
-      .unwrap_or(String::from("Unknown errno"))
+    CStr::from_ptr(libc::strerror(errno)).to_string_lossy().into_owned()
   }
 }
 
 pub fn errno() -> c_int {
   unsafe { *libc::__errno_location() }
 }
 
+macro_rules! dump_errno_str {
+  () => (format!("errno: {}, errno str {}", errno(), strerror(errno())))
```

Comment:
You probably want to fully qualify `errno` and `strerror` this if you intend to use it outside the module ie:

Example from my cuke stuff

```rust
#[macro_export]
macro_rules! Given {
  ($cuke:expr, $regex:expr, $body:expr) => {{
    use $crate::cucumber_regex;
    # Implementation omitted
  }}
}
```